src/aggregate/aggregate_request.c

```diff
@@ -1189,12 +1192,54 @@ static int ApplyVectorQuery(AREQ *req, RedisSearchCtx *sctx, QueryAST *ast, Quer
     QueryNode_ApplyAttributes(vecNode, pvd->attributes, array_len(pvd->attributes), status);
   }
 
-  QueryNode_AddChild(vecNode, ast->root);
+  // Set vector node as ast->root and use setFilterNode for proper filter integration
+  // setFilterNode handles both KNN (child relationship) and RANGE (intersection) properly
+  QueryNode *oldRoot = ast->root;
   ast->root = vecNode;
+  SetFilterNode(ast, oldRoot);
 
   return REDISMODULE_OK;
 }
 
+// Add context to error messages if applicable (currently handles hybrid query context)
+static void addErrorContext(AREQ *req, QueryError *status) {
+  if (QueryError_GetCode(status) == QUERY_OK) {
+    return;
+  }
+
+  QEFlags reqFlags = AREQ_RequestFlags(req);
+
+  // Check if this is a hybrid subquery
+  bool isHybridVectorSubquery = reqFlags & QEXEC_F_IS_HYBRID_VECTOR_AGGREGATE_SUBQUERY;
+  bool isHybridSearchSubquery = reqFlags & QEXEC_F_IS_HYBRID_SEARCH_SUBQUERY;
+
+  if (!isHybridVectorSubquery && !isHybridSearchSubquery) {
+    return; // Not a hybrid query, keep original error
+  }
+
+  RS_ASSERT (isHybridVectorSubquery ^ isHybridSearchSubquery);
+  QueryErrorCode currentCode = QueryError_GetCode(status);
+
+  if (currentCode == QUERY_EVECTOR_NOT_ALLOWED) {
+    // Enhance generic vector error with hybrid context
+    QueryError_ClearError(status);
+    if (isHybridVectorSubquery) {
+      QueryError_SetWithoutUserDataFmt(status, QUERY_EVECTOR_NOT_ALLOWED,
+                                       "Vector queries are not allowed in FT.HYBRID VSIM subquery FILTER");
```

Comment:
I think the use of query and subquery is confusing.

```suggestion
                                       "Specifying vector values are not allowed in FT.HYBRID VSIM sub-query FILTER");
```

Reply:
Removed 'subquery' from the massage, I think its clear now